src/app/qgisapp.cpp

```diff
@@ -2106,6 +2106,13 @@ void QgisApp::dropEvent( QDropEvent *event )
   connect( timer, &QTimer::timeout, this, [this, timer, files, lst]
   {
     QgsCanvasRefreshBlocker refreshBlocker;
+    QgsTemporaryCursorOverride cursorOverride( Qt::WaitCursor );
+    qApp->processEvents();
```

Comment:
Is this explicitly needed? There's a high risk of issues introduced by this.

Reply:
> Is this explicitly needed? There's a high risk of issues introduced by this.

not strictly needed but nice to have the busy cursor as dropping can be quite long. I'll remove it if it isn't safe (I believe I've seen that pattern else where in that file)
It would be nice to have a reliable way of setting the busy cursor and make sure it is well taken into account.

Reply:
removed

Reply:
I was referring to the processEvents, not the cursor override. Did that require the processEvents to take effect?

Reply:
> Did that require the processEvents to take effect?

yes, I tried without, but it had no effect

Reply:
Darn... still, better safe than sorry!